app/src/lib/app-state.ts

```diff
@@ -838,3 +847,112 @@ export function isCherryPickConflictState(
 ): conflictStatus is CherryPickConflictState {
   return conflictStatus.kind === 'cherryPick'
 }
+
+/**
+ * Tracks the state of the app during a multi commit operation such as rebase,
+ * cherry-picking, and interactive rebase (squashing, reordering).
+ *
+ * Depending on the operation or state of the operation, properties may be null.
+ */
+export interface IMultiCommitOperationState {
+  /**
+   * The current step of the operation the user is at.
+   * Examples: ChooseBranchStep, ChooseBranchStep, ShowConflictsStep, etc.
+   */
+  readonly step: MultiCommitOperationStep | null
+
+  /**
+   * The kind of operation it is.
+   * Examples: Rebase, Cherry-pick, Squash, etc.
+   */
+  readonly operationKind: MultiCommitOperationKind | null // rebase, cherry-pick, squash
+
+  /**
+   * The underlying parsed Git information associated with the progress of the
+   * current operation.
+   *
+   * Example: During cherry-picking, after each commit this progress will be
+   * updated to reflect the next commit in the list to cherry-pick.
+   */
+  readonly progress: IMultiCommitOperationProgress | null
+
+  /**
+   * Whether the user has done work to resolve any conflicts as part of this
+   * operation, and therefore, should be warned on aborting the operation.
+   */
+  readonly userHasResolvedConflicts: boolean
+
+  /**
+   * Array of commits used during the operation.
+   */
+  readonly commits: ReadonlyArray<Commit> | null
+
+  /**
+   * If operation specifies a commit that it takes place around
+   *
+   * Example: Squashing all the 'commits' array onto the 'targetCommit'
+   */
+  readonly targetCommit: Commit | null
+
+  /**
+   * If an operation needs can specify a commit message
+   *
+   * Example: Squashing - proving a new commit message/description
+   */
+  readonly commitContext: ICommitContext | null
+
+  /**
+   * For use with interaction rebase operations
+   */
+  readonly lastRetainedCommitRef: string | null
+
+  /**
+   * This is the commit ID of the HEAD of the in-flight operation used to compare
```

Comment:
With commit ID you mean SHA, right?

Reply:
Yes. I copied the description from elsewhere; but can definitely update it since that is what it is. :D